encodings/dict/src/array.rs

```diff
@@ -97,8 +102,25 @@ impl CanonicalVTable<DictArray> for DictEncoding {
                 let canonical_values: Array = array.values().into_canonical()?.into_array();
                 take(canonical_values, array.codes())?.into_canonical()
             }
-            // Non-string case: take and then canonicalize
-            _ => take(array.values(), array.codes())?.into_canonical(),
+            DType::Primitive(ptype, _)
+                // TODO(alex): handle nullable codes & values
+                if array.values().len() <= 32 && *ptype != PType::F16
```

Comment:
How did you choose 32?

Reply:
The max lane count would be 64. Though for large types the amount of allocated stack memory gets respectively large. u64 with 64 lanes = 4096 bytes which is half the max allowed stack size on macOS. It still might be worth considering going for the max lane count here eventually.

Reply:
Checking the signature of the function this should work correctly for any size of values

Reply:
Just ran the tests, indeed works. Very nice! 👍